foreman/data_refinery_foreman/foreman/management/commands/test_rerun_salmon_old_samples.py

```diff
@@ -0,0 +1,288 @@
+import datetime
+import json
+import os
+
+from django.test import TestCase
+from django.utils import timezone
+from typing import Dict, List
+from unittest.mock import MagicMock, Mock, patch, call
+
+from data_refinery_common.job_lookup import ProcessorPipeline, Downloaders
+from data_refinery_common.models import (
+    ComputationalResult,
+    ComputationalResultAnnotation,
+    ComputedFile,
+    DownloaderJob,
+    DownloaderJobOriginalFileAssociation,
+    Experiment,
+    ExperimentOrganismAssociation,
+    ExperimentResultAssociation,
+    ExperimentSampleAssociation,
+    OriginalFile,
+    OriginalFileSampleAssociation,
+    Organism,
+    OrganismIndex,
+    Processor,
+    ProcessorJob,
+    ProcessorJobOriginalFileAssociation,
+    Sample,
+    SampleResultAssociation,
+)
+from data_refinery_foreman.foreman.management.commands.rerun_salmon_old_samples import update_salmon_all_experiments
+
+def setup_experiment(complete_accessions: List[str], incomplete_accessions: List[str]) -> Dict:
+    """ Create an experiment where some samples were processed with the newest version of salmon and
+    other with an older one.
+    """
+    # Create the experiment
+    experiment_accession = 'SRP095529'
+    data_dir = '/home/user/data_store/'
+    experiment_dir = data_dir + experiment_accession
+    experiment = Experiment.objects.create(
+        accession_code=experiment_accession,
+        organism_names=['HOMO_SAPIENS'],
+        technology='RNA-SEQ'
+    )
+
+    zebrafish = Organism.get_object_for_name("DANIO_RERIO")
+
+    # Create the transcriptome processor and result:
+    transcriptome_processor = Processor()
+    transcriptome_processor.name = "Transcriptome"
+    transcriptome_processor.version = "v9.9.9"
+    transcriptome_processor.docker_image = "dr_transcriptome"
+    transcriptome_processor.environment = '{"some": "environment"}'
+    transcriptome_processor.save()
+    computational_result_short = ComputationalResult(processor=transcriptome_processor)
+    computational_result_short.save()
+
+    organism_index = OrganismIndex()
+    organism_index.index_type = "TRANSCRIPTOME_SHORT"
+    organism_index.organism = zebrafish
+    organism_index.result = computational_result_short
+    organism_index.absolute_directory_path = "/home/user/data_store/ZEBRAFISH_INDEX/SHORT"
+    organism_index.salmon_version='v0.9.0'
```

Comment:
This version doesn't match the version of the tx processor which was "used to create" it